src/uucore/src/lib/features/selinux.rs

```diff
@@ -403,4 +503,156 @@ mod tests {
+                assert!(
+                    contexts_differ(path1, path2),
+                    "Contexts should differ when different contexts are set"
+                );
+            }
+        } else {
+            println!(
+                "Note: Couldn't set SELinux contexts to test differences. This is expected if the test doesn't have sufficient permissions."
+            );
+            contexts_differ(path1, path2);
+        }
+
+        let nonexistent_path = Path::new("/nonexistent/file/path");
+        assert!(
+            contexts_differ(path1, nonexistent_path),
+            "contexts_differ should return true when one path doesn't exist"
+        );
+    }
+
+    #[test]
+    fn test_preserve_security_context() {
+        let source_file = NamedTempFile::new().expect("Failed to create source tempfile");
+        let dest_file = NamedTempFile::new().expect("Failed to create destination tempfile");
+        let source_path = source_file.path();
+        let dest_path = dest_file.path();
+
+        std::fs::write(source_path, b"source content").expect("Failed to write to source tempfile");
+        std::fs::write(dest_path, b"destination content")
+            .expect("Failed to write to destination tempfile");
+
+        if !is_selinux_enabled() {
+            let result = preserve_security_context(source_path, dest_path);
+            assert!(
+                result.is_ok(),
+                "preserve_security_context should succeed when SELinux is not enabled"
+            );
+            return;
+        }
+
+        let source_context = String::from("system_u:object_r:tmp_t:s0");
+        let result = set_selinux_security_context(source_path, Some(&source_context));
+
+        if result.is_ok() {
+            let preserve_result = preserve_security_context(source_path, dest_path);
+            assert!(
+                preserve_result.is_ok(),
+                "Failed to preserve context: {:?}",
+                preserve_result.err()
+            );
+
+            assert!(
+                !contexts_differ(source_path, dest_path),
+                "Contexts should be the same after preserving"
+            );
+
+            let dest_context =
+                get_selinux_security_context(dest_path).expect("Failed to get destination context");
+            assert!(
+                dest_context.contains("tmp_t"),
+                "Expected destination context to contain 'tmp_t', but got: {}",
+                dest_context
+            );
+        } else {
+            println!(
+                "Note: Couldn't set SELinux context on source file to test preservation. This is expected if the test doesn't have sufficient permissions."
+            );
+
+            let preserve_result = preserve_security_context(source_path, dest_path);
+            if let Err(err) = preserve_result {
+                println!("Context preservation failed as expected: {}", err);
+            }
```

Comment:
Here I would use an `assert!`:
```suggestion
            assert!(preserve_result.is_err());
```